lib/commandCenter/commandCenter.js

```diff
@@ -388,15 +387,52 @@ function compress(content, cb) {
 	});
 }
 
+/**
+ * Promisify the user command execute method, or
+ * ensure its return value is passed to state.respond
+ *
+ * Once callback-based user commands are removed from MAGE,
+ * you will be able to remove this method.
+ *
+ * @param {*} cmdInfo
+ * @param {*} paramList
+ */
+function generateCommandPromise(cmdInfo, state, paramList) {
+	const mod = cmdInfo.mod;
+	const execute = mod.execute.apply.bind(mod.execute, mod, paramList);
```

Comment:
Should this not be `..paramList`? And if so, I'm a bit worried tests didn't catch that.

Reply:
No, because we cannot use destructuring since we need to support Node 4.

This is why we `bind` to `apply(this, argsArray)`  instead of on `mod.execute` directly.

Reply:
Oh wow. Okay. `apply` *and* `bind`... Alright.

Reply:
*Yeah...* suggestions welcome though, this looks... bad.

Reply:
Suggestions...

Either a lambda:

```js
const execute = () => {
  mod.execute(...paramList);
};
```

Or simply not bothering with a shortcut at all:

```js
	if (cmdInfo.isAsync) {
		// Async user commands will need to take the response value, and feed it
		// to `state.respond`
		return mod.execute(...paramList)
			.then((response) => state.respond(response, mod.serialize === false));
	} else {
		// Callback-based user commands are expected to call `state.respond`
		// on their own; all we need to do is promisify
		return new Promise((resolve) => mod.execute(...paramList, resolve));
	}
```

Reply:
The first option might be more readable. I'm sold.